docs/reference/language/context.md

```diff
@@ -25,38 +25,201 @@ in some places that are also aware of their location in the document:
 [Show rules]($styling/#show-rules) provide context[^1] and numberings in the
 outline, for instance, also provide the proper context to resolve counters.
 
-## Style context
-With set rules, we can adjust style properties for parts or the whole of our
-document. We cannot access these without a known context, as they may change
-throughout the course of the document. When context is available, we can
-retrieve them simply by accessing them as fields on the respective element
-function.
+## Behavior of the context keyword
+Style properties frequently change within a document, for example by applying set 
+rules. To retrieve such properties in a consistent way, one must first specify 
+the precise context where the property should be retrieved. This can be achieved 
+with the `context` keyword. Once the context has been fixed, the property 
+information is available through standard field access syntax. For example, 
+`text.lang` asks for the current language setting. In its simplest form, the 
+`context` keyword refers to "right here":
 
 ```example
 #set text(lang: "de")
+// read the language setting "here"
 #context text.lang
 ```
 
-As explained above, a context expression is reactive to the different
-environments it is placed into. In the example below, we create a single context
-expression, store it in the `value` variable and use it multiple times. Each use
-properly reacts to the current surroundings.
+Note that any attempt to access `#text.lang` directly, i.e. outside of a context,
+will cause the compiler to issue an error message. The field names supported 
```

Comment:
I think it's worth adding an extra sentence here explaining why the compiler has no choice but the emit an error: Because it needs to know at which location you want the property.